hydra-node/src/Hydra/Party.hs

```diff
@@ -0,0 +1,151 @@
+{-# LANGUAGE TypeApplications #-}
+-- ToJSON VerificationKey
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- | Types and functions revolving around a Hydra 'Party'. That is, a
+-- participant in a Hydra Head, which signs transactions or snapshots in the
+-- Hydra protocol. Hence, this module also includes functions to sign and verify
+-- data given a 'Party'.
+module Hydra.Party where
+
+import Hydra.Prelude hiding (show)
+
+import Cardano.Crypto.DSIGN (
+  DSIGNAlgorithm (..),
+  MockDSIGN,
+  SigDSIGN (..),
+  SignKeyDSIGN,
+  VerKeyDSIGN,
+  signDSIGN,
+ )
+import Cardano.Crypto.Seed (mkSeedFromBytes)
+import Cardano.Crypto.Util (SignableRepresentation)
+import Data.Aeson (ToJSONKey, Value (String), object, withText, (.=))
+import Data.Aeson.Types (FromJSONKey)
+import qualified Data.ByteString.Base16 as Base16
+import Test.QuickCheck (vectorOf)
+import Text.Show (Show (..))
+
+-- | Identifies a party in a Hydra head by it's 'VerificationKey'.
+data Party = Party
+  { alias :: Maybe Text
+  , vkey :: VerificationKey
+  }
+  deriving (Eq, Generic, FromJSON)
+
+instance Show Party where
+  show Party{alias, vkey} =
+    toString $ prefix <> showVerificationKey vkey
+   where
+    prefix = case alias of
+      Nothing -> ""
+      Just a -> a <> "@"
+
+-- NOTE(SN): This instance is ordering aliased parties in front of unaliased
+-- parties. It is covered well by tests, but I don't feel too comfortable in
+-- implementing such low-level instances, which could impact 'Set' behavior.
+instance Ord Party where
```

Comment:
Sugggestion: We should property test equivalence of `compare a b == EQ` and `a == b`.

Reply:
What do you think of the suggested https://hackage.haskell.org/package/quickcheck-classes-0.6.5.0/docs/Test-QuickCheck-Classes.html#v:ordLaws?

Reply:
Done

Reply:
Yes, using quickcheck-classes could simplify the code, removing "manual" properties instances. Definitely a refactoring thing for later.